ethers-core/src/abi/mod.rs

```diff
@@ -117,7 +117,7 @@ macro_rules! impl_abi_type {
 }
 
 impl_abi_type!(
-    Vec<u8> => Bytes,
+    Vec<u8> =>  Array(Box::new(ParamType::Uint(8))),
```

Comment:
I think this won't work, because `bytes memory` which translates to `Vec<u8>` will now be translate to `Array(Uint(8))` as a `uint8[]`.

Could we make `bytes memory` translate to `ethers::core::Bytes`?

(I think that's probably the bug here, given the tests' output)